### ISSUE TITLE
Questions about loss and likelihood implementation

### ISSUE THREAD

**Issue by michaelGuo1204** [raphael-group/GASTON#10]
Hi!

Your work on integrating spatial gradients and multiple tasks in ST data analysis is truly inspiring!

I'm opening this issue to for some tiny problems in GASTON framework. According to your article, a neural network is used to predict the expression \(a_{i}\) from spatial coordinates \((x,y)\) via isodepth \(d_i\). In the methods section (lines 613-616) of the preprint, it states that "solving (10) is equivalent to learning the parameters of a single network...". My understanding is that this implies using the likelihood of \(\mathbb{P}(a_{i,g}|h_{\theta'(d_{\theta}(x,y))})\) as the training loss for the network. However, while reading the code, I noticed that the `gaston.neural_net.train` function uses a general MSE loss to evaluate the bias between the reconstructed expression profile and the ground truth GLM-PCs. Is this the intended behavior?

https://github.com/raphael-group/GASTON/blob/08fdfc2297724e79d26e72074d53644f8cbb0064/src/gaston/neural_net.py#L167

Additionally, I observed that a linear regression, rather than a Poisson regression, is applied during segment regression. Does this mean that a simple linear regression is sufficient for handling the expression profiles?

https://github.com/raphael-group/GASTON/blob/08fdfc2297724e79d26e72074d53644f8cbb0064/src/gaston/dp_related.py#L61

Looking forward to your response!

Best regards!
14 Dec 2024

**Comment by uthsavc** [comment]
Hi, thanks for your interest in our work! So sorry for my delay in answering. 

For the first question, yes we intend to use the MSE loss for the Poisson PCs. This is because, for data following a Poisson distribution with P domains, the top-2P GLM-PCs are approximately normally distributed, and so it is (approximately) equivalent to use a Poisson regression loss with the raw count data, versus a MSE loss with the top-2P GLM-PCs. See Sec 4.4 of manuscript https://www.biorxiv.org/content/10.1101/2023.10.10.561757v1.full.pdf (Though in practice you may sometimes want to use more or less than 2P GLM-PCs, especially if you don't know what P is)

For the second question, we have two phases of segmented regression. When we are using the GLM-PCs, we use a least squares/MSE loss (see above). When we use individual genes (ie for finding spatially varying genes), then we use a Poisson loss.

Hope this helps!

**Comment by michaelGuo1204** [comment]
Got it. Really appreciate your reply! ;)
